### PR TITLE
Added Sieve exercise

### PR DESCRIPTION
**Pull request by ryanplusplus**
See:
- [Description](https://github.com/exercism/x-common/blob/master/sieve.md)
- [Test list](https://github.com/exercism/x-common/blob/master/sieve.json)

### CODE REVIEW

**Comment by aarti** [comment]
This exercise is a prime candidate for `coroutines`, What do you think?

**Comment by ryanplusplus** [comment]
Are you suggesting to explicitly require a coroutine to be returned that generates primes?

**Comment by aarti** [comment]
Yes, that would be great, just for this exercise, seems like a standard way to teach coroutines http://www.cs.dartmouth.edu/~doug/sieve/  http://www.lua.org/cgi-bin/demo

**Comment by ryanplusplus** [comment]
Okay sounds good to me. I should be able to get this updated in the next day or so. Thanks for the tip!

**Comment by ryanplusplus** [comment]
Hey @aarti I refactored to use coroutines and the implementation cleaned up very nicely. Can you have another look?

File: config.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "leap",
     "space-age",
     "pascals-triangle",
+    "sieve",
```
Comment (ryanplusplus) [inline config.json]: I moved this a little bit further down to give people more time to get familiar with Lua before tackling coroutines.

**Comment by aarti** [comment]
:+1:

**Pull request merged**